### PR TITLE
Maintain active dataset fields

### LINKED ISSUE #1852: [BUG] Changing session.view resets field visibility choices
On `fiftyone==0.16.2`, updating `session.view` resets any field visibility toggles I may have set (eg, unselected all label fields), and forces the defaults (all label fields visible). I don't think this used to be the case though?

This came up when I was trying to work with an interactive plot. I just wanted to see images with no labels, but every time I made a selection in the linked plot, the labels kept re-appearing, which was annoying. I don't recall facing this issue before.

Of course persisting sidebar settings is a bit tricky because views can change the label schema.

### PR DESCRIPTION
**Pull request by benjaminpkane**
Resolves #1852

### CODE REVIEW

File: app/packages/app/src/Root/Datasets/Dataset.tsx [hunk 1]
```diff
@@ -97,17 +98,24 @@ export const Dataset: Route<DatasetQuery> = ({ prepared }) => {
...
-    update(({ reset }) => {
+    update(({ reset, get }) => {
       reset(filters);
-      reset(_activeFields({ modal: false }));
       reset(similarityParameters);
 
+      const newDataset = transformDataset(dataset);
+
+      const oldDataset = get(atoms.dataset);
+      oldDataset && console.log(oldDataset.id, newDataset.id);
```
Comment (ritch) [inline app/packages/app/src/Root/Datasets/Dataset.tsx:108]: Stray console.log?

Comment (benjaminpkane) [reply app/packages/app/src/Root/Datasets/Dataset.tsx:108]: Yes, thanks

File: app/packages/app/src/Root/Datasets/Dataset.tsx [hunk 2]
```diff
@@ -97,17 +98,24 @@ export const Dataset: Route<DatasetQuery> = ({ prepared }) => {
...
+      if (!oldDataset || oldDataset.id !== newDataset.id) {
+        reset(_activeFields({ modal: false }));
```
Comment (ritch) [inline app/packages/app/src/Root/Datasets/Dataset.tsx:108]: Is there a reason this isn't in `useStateUpdate()`? Looks like there are similar things there.

**Review by ritch — APPROVED** [review]
FIx LGTM. Left a few comments, but shouldn't hold up the fix.

File: app/packages/app/src/Root/Datasets/Dataset.tsx (hunk 2 above)
Comment (benjaminpkane) [reply app/packages/app/src/Root/Datasets/Dataset.tsx:108]: Good point, fixed

**Pull request merged**